markdown/events/2022/bytesize-gitpod.md

```diff
@@ -25,3 +25,121 @@ It is our hope that these talks / videos will build an archive of training mater
+
+[5:02](https://www.youtube.com/watch?v=kBoC6QBU-M0t=302)
+What we're going to do now is, we're just going to go straight into how to use Gitpod. Hopefully everyone can see this. The example pipeline we're going to use is the nf-core test pipeline. I've already got the browse extension so I actually have this link already. I could also copy the link address, open up a new tab and just paste it in. But for speed I'll just click on it. So when you log in, all you need to do... and at this point it would ask you for your credentials. For me it's not because I'm already logged in. This is going straight from that repository into Gitpod. Along on the other side you can see the explorer set says, these are all the files within your repository; you have a window here to look at visualization of code or PDFs etc and then you have a terminal at the bottom. You also should have extra buttons here which I can't see. Anyway, there's all the things you would normally want to do on your own computer, you can also do here.
+
+[6:21](https://www.youtube.com/watch?v=kBoC6QBU-M0t=381)
+Just to show you how this is actually working, the Gitpod.yaml file is not... Did I click the right one? I'll go back. This particular repository is in the master. For some pipelines (and this is a very good example), that you want to run on Gitpod, you need to make sure it's got this Gitpod.yaml file, because this is all the programs you need. If we click on that, it's showing that we have an image, so a docker file from dockerHub. The image is being pulled, that has all the code that we need and I will go a bit more into these Gitpod yamls a bit later.
+
+[7:05](https://www.youtube.com/watch?v=kBoC6QBU-M0t=425)
+We need to be in this development branch. I also mention at this point, if you're actually making changes to nf-core code then you'll be wanting to do this in the dev branch and not on the master branch. If we click it, we should get back to a Gitpod environment, that I was expecting. That should have all the code that we need to run this particular pipeline. Sometimes this can take a few seconds. Probably because it's a live demo, it'll be more than that, but if not we can move on to different things I can explain to you. It should pop up reasonably fast. It won't? No. Okay, there we go. Now we are actually in the the correct repository that I wanted to be in, but just move that away, just to make it clearer. You can see there's a gitpod.yml file which I was expecting to see. This is the reason why we have the code brought into this environment. In the terminal, if we type in `nextflow` it finds the tool because it's already installed within this gitpod.yml file. It told us that this container contains all the things we need to run the particular piece of code.
```

Comment:
```suggestion
We need to be in this development branch. I also mention at this point, if you're actually making changes to nf-core code then you'll be wanting to do this in the dev branch and not on the master branch. If we click it, we should get back to a Gitpod environment, that I was expecting. That should have all the code that we need to run this particular pipeline. Sometimes this can take a few seconds. Probably because it's a live demo, it'll be more than that, but if not we can move on to different things I can explain to you. It should pop up reasonably fast. It won't? No. Okay, there we go. Now we are actually in the correct repository that I wanted to be in, but just move that away, just to make it clearer. You can see there's a gitpod.yml file which I was expecting to see. This is the reason why we have the code brought into this environment. In the terminal, if we type in `nextflow` it finds the tool because it's already installed within this gitpod.yml file. It told us that this container contains all the things we need to run the particular piece of code.
```